xbmc/video/VideoDatabase.cpp

```diff
@@ -4304,23 +4310,12 @@ void CVideoDatabase::GetCast(int media_id, const std::string &media_type, std::v
     {
       SActorInfo info;
       info.strName = m_pDS2->fv(0).get_asString();
-      bool found = false;
-      for (const auto &i : cast)
-      {
-        if (i.strName == info.strName)
-        {
-          found = true;
-          break;
-        }
-      }
-      if (!found)
-      {
-        info.strRole = m_pDS2->fv(1).get_asString();
-        info.order = m_pDS2->fv(2).get_asInt();
-        info.thumbUrl.ParseString(m_pDS2->fv(3).get_asString());
-        info.thumb = m_pDS2->fv(4).get_asString();
-        cast.emplace_back(std::move(info));
-      }
+      info.strRole = m_pDS2->fv(1).get_asString();
+      info.order = m_pDS2->fv(2).get_asInt();
+      info.thumbUrl.ParseString(m_pDS2->fv(3).get_asString());
+      info.thumb = m_pDS2->fv(4).get_asString();
+      cast.emplace_back(std::move(info));
+
```

Comment:
How come you no longer need to check of the actor is found?

Reply:
Before the logic actually made sure thet the same actor does not appear twice in the `cast` variable by only adding a new actor if it isn't already part of the cast. This is exactly what I want to change. I have no idea why the manual logic was there in the first place because it was already prevented by the `UNIQUE INDEX` but maybe that one was added later and someone forgot to remove the manual logic here.

Reply:
I think that adding to my confusion 😉